app/controllers/search.js

```diff
@@ -34,59 +37,103 @@ export default Ember.Controller.extend(backNavigator, {
     return Ember.$.trim(this.get("searchText")).length;
   }),
 
-  hasFilter: Ember.computed("filter", function() {
-    return Ember.$.trim(this.get("filter")).length;
-  }),
-
   onSearchTextChange: Ember.observer("searchText", function() {
-    // wait before applying the filter
-    Ember.run.debounce(this, this.applyFilter, 500);
+    const searchTextLength = this.get("searchText").length;
+    if (
+      searchTextLength > this.get("minSearchTextLength") ||
+      searchTextLength === 0
+    ) {
+      this.reloadResults();
+    }
   }),
 
-  applyFilter: function() {
-    this.set("filter", this.get("searchText"));
-    this.searchOnServer();
+  on() {
+    this.showResults(); // Upon opening the page, we populate with results
+    this.get("filterService").on("change", this, this.reloadResults);
   },
 
-  searchOnServer() {
-    let search = this.get("filter");
-    if (!search) {
-      this.set("filteredResults", null);
-      return;
-    }
+  off() {
+    this.get("filterService").off("change", this, this.reloadResults);
+  },
+
+  reloadResults() {
+    this.hideResults();
+    Ember.run.debounce(this, this.showResults, 500);
+  },
+
+  hideResults() {
+    Ember.run(() => {
+      this.set("displayResults", false);
+    });
+  },
+
+  showResults() {
+    Ember.run(() => {
+      this.set("displayResults", true);
+    });
+  },
 
-    let loadingView = getOwner(this)
-      .lookup("component:loading")
-      .append();
-    let url = `/offers/search?searchText=${search}`;
-    let store = this.get("store");
-
-    new AjaxPromise(url, "GET", this.get("session.authToken"))
-      .then(data => {
-        store.pushPayload(data);
-        const results = data.offers
-          .map(o => store.peekRecord("offer", o.id))
-          .filter(Boolean);
-        this.set("filteredResults", results);
-      })
-      .finally(() => {
-        loadingView.destroy();
-      });
+  getFilterQuery() {
+    const filterService = this.get("filterService");
+    const isPriority = filterService.isPriority();
+    const { after, before } = filterService.get("offerTimeRange");
+    let stateFilters = _.without(
```

Comment:
I don't think `"showPriority"` should be mixed with the state filters.
It doesn't make sense and forces those kinds of workarounds. `showPriority` should be a boolean we toggle.

This is copied from stock, no need to change it now.
But if anyone has an opportunity/bandwidth one day it'd be nice to clear it up :) 

cc - @abhinav098 @namrataukirde @abulsayyad123